src/Microsoft.AspNetCore.Authentication/AuthenticationBuilder.cs

```diff
@@ -97,6 +97,10 @@ public EnsureSignInScheme(IOptions<AuthenticationOptions> authOptions)
             public void PostConfigure(string name, TOptions options)
             {
                 options.SignInScheme = options.SignInScheme ?? _authOptions.DefaultSignInScheme;
+                if (options.SignInScheme == name)
+                {
+                    throw new InvalidOperationException(Resources.Exception_RemoteSignInSchemeCannotBeSelf);
```

Comment:
Note: this is technically in the wrong place, it SHOULD be in RemoteAuthenticationOptions.Validate....but we don't pass in the authenticationScheme name we are validating :(  whoops...   So this is the next best spot, its almost as good (but will block anyone from setting sign in scheme via a later PostConfigure)